### DISCUSSION TITLE
Add window resizing like hyprland has.

### DISCUSSION THREAD

**Discussion by IVKLD** [YaLTeR/niri#478] [Ideas]
On 4k monitors, there is severe lag when resizing a window, whether it's firefox or a text editor. I would like to see an implementation of window resizing like in hyprland. In norms, when resizing a window, it changes the size of the frame, which is not bound to the application window.

**Comment by YaLTeR** [comment]
The lag comes from the client itself. Firefox in particular uses CPU rendering for parts of the surface (CSD shadow maybe) from GTK 3, that's where the slowness comes from.

I'm not a big fan of clipping windows to their frame all the time because:
1. With `clip-to-geometry false` (the default) it will straight away clip out parts of the buffer (like CSD shadows), which will look janky. Hyprland always uses a close equivalent of `clip-to-geometry true` so they can get away with it.
2. Usual tiling WMs always have the restriction that windows must add up to the width of the monitor, so they aggressively use clipping to achieve that when the clients end up with a different size. Scrollable tiling's infinite horizontal space lifts the width part of this restriction, so I can avoid clipping in niri and always show the correct full window contents.

I guess maybe some kind of option for this could be added, but it may require nontrivial changes to the layout. Interactive resizing is among the most complex interactions due to the asynchrony involved.

**Comment by IVKLD** [comment]
Got it. Will this be added as a separate option?

**Reply by YaLTeR** [reply]
maybe

**Reply by IVKLD** [reply]
Благодарю
